android/src/main/java/com/sensors/RNSensor.java

```diff
@@ -46,11 +46,9 @@ public RNSensor(ReactApplicationContext reactContext, String sensorName, int sen
   @ReactMethod
   public void isAvailable(Promise promise) {
     if (this.sensor == null) {
-      // No sensor found, throw error
-      promise.reject(new RuntimeException("No " + this.sensorName + " found"));
```

Comment:
It is likely this RuntimeException() that kills the process. Do we know if the other changes are absolutely necessary?